main.js

```diff
@@ -167,7 +176,7 @@ function createWindow () {
 
   // Emitted when the window is about to be closed.
   mainWindow.on('close', function (e) {
-    if (process.platform === 'darwin' && !windowState.shouldQuit() && config.environment !== 'test') {
+    if ((usingTrayIcon || process.platform === 'darwin') && !windowState.shouldQuit() && config.environment !== 'test') {
```

Comment:
We should probably break this up into explanatory variables, now that it's gotten so long. Can you do that?

Reply:
I broke it in two meaningful if statements, hope it looks better now.